### PR TITLE
Improve oclint

### PR DESCRIPTION
**Pull request by mgrebenets**
- Update `Actions.md` with latest oclint changes
- Deprecate `select_reqex`, replace with `select_regex`
- Return actual shell command from `self.run` for rspec
- Add `exclude_regex` parameter
- Add `enable_rules` and `disable_rules` parameters
- Add deprecation message for `rc` parameter
- Add 3 new parameters for static analysis configuration
- Add support for array of `thresholds` (multiple -rc)
- Add `list_enabled_rules` parameter

### CODE REVIEW

File: lib/fastlane/actions/oclint.rb [hunk 1]
```diff
@@ -28,11 +37,27 @@ def self.run(params)
         report_path = params[:report_path] ? params[:report_path] : 'oclint_report.' + report_type
 
         oclint_args = ["-report-type=#{report_type}", "-o=#{report_path}"]
-        oclint_args << "-rc=#{params[:rc]}" if params[:rc]
+
+        oclint_args << "-list-enabled-rules" if params[:list_enabled_rules]
+
+        if params[:rc]
+          Helper.log.warn "It's recommended to use 'thresholds' instead of deprecated 'rc' parameter".yellow
```
Comment (KrauseFx) [inline lib/fastlane/actions/oclint.rb]: :heart:

File: lib/fastlane/actions/oclint.rb [hunk 2]
```diff
@@ -40,9 +65,9 @@ def self.run(params)
           '"' + files.join('" "') + '"'
         ].join(' ')
 
-        Action.sh command
-
         Actions.lane_context[SharedValues::FL_OCLINT_REPORT_PATH] = File.expand_path(report_path)
+
+        Action.sh command # Needs to be last as return value for tests
```
Comment (KrauseFx) [inline lib/fastlane/actions/oclint.rb]: You could remove the comment and just add a `return` statement to make it clear. Could you also add `(` `)` around the `command`?

File: lib/fastlane/actions/oclint.rb [hunk 3]
```diff
@@ -60,11 +85,22 @@ def self.available_options
                                        description: 'The json compilation database, use xctool reporter \'json-compilation-database\'',
                                        default_value: 'compile_commands.json',
                                        optional: true),
+          # Deprecated, remove as soon as possible
```
Comment (KrauseFx) [inline lib/fastlane/actions/oclint.rb]: Everything below or just this one option? If just this one option, this should be clearer

**Comment by KrauseFx** [comment]
This is great, thank you for this PR. Could you review my comments, and squash all your commits to one? Please merge back from master to get the `rubocop` fixes so that the Travis build passes :+1:

File: lib/fastlane/actions/oclint.rb (hunk 3 above)
Comment (mgrebenets) [inline lib/fastlane/actions/oclint.rb]: Moved it to the next line to be more specific.

File: lib/fastlane/actions/oclint.rb (hunk 2 above)
Comment (mgrebenets) [inline lib/fastlane/actions/oclint.rb]: Yep, done.

**Comment by KrauseFx** [comment]
Thank you so much @mgrebenets :+1:

**Pull request merged**